thread/zos390/rasthrsup.c

```diff
@@ -26,5 +26,6 @@
 uintptr_t
 omrthread_get_ras_tid(void)
 {
-	return (uintptr_t)(*(uint32_t *)pthread_self().__);
+	pthread_t id = pthread_self();
+	return (uintptr_t)*(unsigned long long *)&id;
```

Comment:
how does the cast `(unsigned long long *)` help?

Reply:
`pthread_t` is 8 bytes on z/OS (even in 31-bit mode); the system header uses `unsigned long long` (for some part of `__`).